### PR TITLE
RHDM-1171: update npm deps, rework linting and ts config, adjust code

### PR DESCRIPTION
**Pull request by LoneRifle**
This PR implements the dependency updates found in #380 so that they can be merged
to master while the select refactor is still underway. Merging this first would allow downstream forks to pick this up for follow-on work on the frontend

### CODE REVIEW

File: optaweb-employee-rostering-frontend/cypress/integration/newTenantBasicWorkflow.js [hunk 1]
```diff
@@ -15,9 +15,9 @@
  */
 
 function selectValue(selectPlaceholder, value) {
-  cy.get(`[placeholder="${selectPlaceholder}"]`).clear();
-  cy.get(`[placeholder="${selectPlaceholder}"]`).type(value);
-  cy.get('button').contains(value).last().click();
+  cy.get(`[aria-label="${selectPlaceholder}"]`).clear({ force: true });
+  cy.get(`[aria-label="${selectPlaceholder}"]`).type(value, { force: true });
+  cy.get('div').contains(value).last().click({ force: true });
```
Comment (Christopher-Chianelli) [inline optaweb-employee-rostering-frontend/cypress/integration/newTenantBasicWorkflow.js]: This is from the Select refactor; the change will break integration tests without the Select refactor.

File: optaweb-employee-rostering-frontend/src/ui/pages/shift/EditShiftModal.tsx [hunk 2]
```diff
@@ -189,11 +190,12 @@ export class EditShiftModal extends React.Component<Props & WithTranslation, Sta
               emptyText={t('none')}
               value={(this.state.editedValue.rotationEmployee !== null)
                 ? this.state.editedValue.rotationEmployee : undefined}
-              options={[undefined, ...this.props.employeeList]}
```
Comment (Christopher-Chianelli) [inline optaweb-employee-rostering-frontend/src/ui/pages/shift/EditShiftModal.tsx]: This is from the select refactor

File: optaweb-employee-rostering-frontend/src/ui/pages/shift/EditShiftModal.tsx [hunk 3]
```diff
@@ -175,11 +175,12 @@ export class EditShiftModal extends React.Component<Props & WithTranslation, Sta
               emptyText={t('unassigned')}
               value={(this.state.editedValue.employee !== null)
                 ? this.state.editedValue.employee : undefined}
-              options={[undefined, ...this.props.employeeList]}
```
Comment (Christopher-Chianelli) [inline optaweb-employee-rostering-frontend/src/ui/pages/shift/EditShiftModal.tsx]: This is from the select refactor

**Comment by Christopher-Chianelli** [comment]
Also, RHDM-1171 is the title of the issue about Select options being hidden when inside Modals; you should change it (in both the PR and the commit, if you are removing the Select refactor).

**Comment by LoneRifle** [comment]
Given discussion here and in the other PR, I'm happy to close this

**Pull request closed without merging**